src/modules/idyntree-yarp-visualizer/RobotConnectors.cpp

```diff
@@ -0,0 +1,558 @@
+                     "followed by the full list of joints in the control board.";
+        return false;
+    }
+
+    for (size_t i = 0; i < cbJointList->size(); i += 2)
+    {
+        m_cb_jointsMap.emplace_back();
+        m_cb_jointsMap.back().jointsToConsider = -1;
+        std::vector<JointInfo>& jointVector = m_cb_jointsMap.back().joints;
+
+        yarp::os::Value& cbName = cbJointList->get(i);
+        if (cbName.isString())
+        {
+            m_cb_jointsMap.back().name = cbName.asString();
+
+        }
+        else if (cbName.isList())
+        {
+            yarp::os::Bottle* cbNameList = cbName.asList();
+            if (cbNameList->size() != 2 && cbNameList->size() != 1)
+            {
+                yError() << "The element in position" << i << "(0-based) of the connectToStateExt list is a malformed list. It is supposed to be one or two dimensional,"
+                            " containing the name of the board and, optionally the number of considered joints";
+                return false;
+            }
+
+            if (!cbNameList->get(0).isString())
+            {
+                yError() << "The element in position" << i << "(0-based) of the connectToStateExt list is a malformed list. The first element is supposed to be a string containing the name of the control board.";
+                return false;
+            }
+
+            m_cb_jointsMap.back().name = cbNameList->get(0).asString();
+
+            if (cbNameList->size() == 2)
+            {
+                if (!cbNameList->get(1).isInt())
+                {
+                    yError() << "The element in position" << i << "(0-based) of the connectToStateExt list is a malformed list. The second element is supposed to be int defining the number of considered joints.";
+                    return false;
+                }
+
+                m_cb_jointsMap.back().jointsToConsider = cbNameList->get(1).asInt();
+            }
+        }
+        else
+        {
+            yError() << "The element in position" << i << "(0-based) of the connectToStateExt list is supposed to be a string or a list, containing the name of the board and, optionally the number of considered joints.";
+            return false;
+        }
+
+        yarp::os::Value& jointListValue = cbJointList->get(i+1);
+        if (!jointListValue.isList())
+        {
+            yError() << "The element in position" << i+1 << "(0-based) of the connectToStateExt list is supposed to be a list, namely the list of joints of the specific control board.";
+            return false;
+        }
+
+        yarp::os::Bottle* jointList = jointListValue.asList();
+
+        for (size_t j = 0; j < jointList->size(); ++j)
+        {
+            yarp::os::Value& jointValue = jointList->get(j);
+
+            if (jointValue.isString())
+            {
+                jointVector.push_back({jointValue.asString(), JointType::REVOLUTE});
```

Comment:
is this default consideration documented?

Reply:
Here https://github.com/robotology/idyntree-yarp-tools/blob/6030bccc08320dd566514dfd72b56edf36ce3d3e/src/modules/idyntree-yarp-visualizer/Visualizer.cpp#L540

Reply:
Great thanks. I noted that, but forgot to update the review.